libsolidity/codegen/CompilerUtils.cpp

```diff
@@ -741,17 +740,43 @@ void CompilerUtils::convertType(
 		else if (targetTypeCategory == Type::Category::FixedPoint)
 		{
 			solAssert(
-				stackTypeCategory == Type::Category::Integer || 
+				stackTypeCategory == Type::Category::Integer ||
 				stackTypeCategory == Type::Category::RationalNumber ||
 				stackTypeCategory == Type::Category::FixedPoint,
 				"Invalid conversion to FixedMxNType requested."
 			);
-			//shift all integer bits onto the left side of the fixed type
+
 			FixedPointType const& targetFixedPointType = dynamic_cast<FixedPointType const&>(_targetType);
-			if (auto typeOnStack = dynamic_cast<IntegerType const*>(&_typeOnStack))
-				if (targetFixedPointType.numBits() > typeOnStack->numBits())
-					cleanHigherOrderBits(*typeOnStack);
-			solUnimplemented("Not yet implemented - FixedPointType.");
+			if (stackTypeCategory == Type::Category::Integer)
+			{
+				IntegerType const& typeOnStack = dynamic_cast<IntegerType const&>(_typeOnStack);
+				cleanHigherOrderBits(typeOnStack);
+				m_context << u256(pow(bigint(10), targetFixedPointType.fractionalDigits()))
+						  << Instruction::MUL;
+				if (_cleanupNeeded) {
+					cleanHigherOrderBits(targetFixedPointType);
+				}
+			}
+			else if (stackTypeCategory == Type::Category::RationalNumber)
+			{
+				auto typeOnStack = dynamic_cast<RationalNumberType const*>(&_typeOnStack);
+				convertType(*typeOnStack->fixedPointType(), targetFixedPointType);
+			}
+			else
+			{
+				solAssert(stackTypeCategory == Type::Category::FixedPoint, "");
+				auto typeOnStack = dynamic_cast<FixedPointType const*>(&_typeOnStack);
+				cleanHigherOrderBits(*typeOnStack);
+
+				if (typeOnStack->fractionalDigits() > targetFixedPointType.fractionalDigits())
+					m_context << u256(pow(bigint(10), typeOnStack->fractionalDigits() - targetFixedPointType.fractionalDigits()))
+							  << Instruction::SWAP1
+							  << (typeOnStack->isSigned() ? Instruction::SDIV : Instruction::DIV);
+				else if (typeOnStack->fractionalDigits() < targetFixedPointType.fractionalDigits())
+					m_context << u256(pow(bigint(10), targetFixedPointType.fractionalDigits() - typeOnStack->fractionalDigits()))
```

Comment:
I liked your scaling helper in `ABIFunctions`. Do you think a helper could be created for this file?